test/fx/test_fx_param_shape_control_flow.py

```diff
@@ -0,0 +1,44 @@
+import unittest
+
+import torch
+import torch.fx
+import difflib
+
+
+class MyModule(torch.nn.Module):
+    def __init__(self, in_channels):
+        super().__init__()
+        self.param = torch.nn.Parameter(torch.randn(in_channels, 3))
+
+    def forward(self, x):
+        if self.param.shape[0] < 10:
+            return torch.mm(x, self.param)
+        else:
+            return torch.relu(torch.mm(x, self.param))
+
+
+class TestConstParamShapeInControlFlow(unittest.TestCase):
+    def test_param_shape_const(self):
+        mymod = MyModule(in_channels=5)
+
+        # Test that we go down the True branch in forward
+        x = torch.randn(10, 5)
+        torch.testing.assert_allclose(mymod(x), torch.mm(x, mymod.param))
+        tracer = torch.fx.Tracer(param_shapes_constant=True)
+        traced_graph = tracer.trace(mymod)
+
+        # Make a new module with different parameter shape to go down the different
+        # code path
+        mymod2 = MyModule(in_channels=15)
+        x = torch.randn(10, 15)
+        torch.testing.assert_allclose(mymod2(x), torch.relu(torch.mm(x, mymod2.param)))
+
+        tracer2 = torch.fx.Tracer(param_shapes_constant=True)
+        traced_graph2 = tracer2.trace(mymod2)
+
+        # the second graph has an exta relu function call node
+        assert(len(traced_graph2.nodes) > len(traced_graph.nodes))
```

Comment:
This is probably fine, but a more robust test might check for the presence of only a `torch.mm` target in the first graph and both the `torch.relu` and `torch.mm` target in the second graph

Reply:
Added check for relu and mm nodes.